pkg/cloudprovider/cloudprovider.go

```diff
@@ -84,7 +89,7 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 	kubeDNSIP, err := kubeDNSIP(ctx, ctx.KubernetesInterface)
 	if err != nil {
-		logging.FromContext(ctx).Debugf("unable to detect the IP of the kube-dns service, %s", err)
```

Comment:
crashing caused customers issues here with custom DNS setups. https://github.com/aws/karpenter/issues/2787

Reply:
Sounds good, I'll add a comment linking the issue because I certainly won't remember this reason down the line.